lib/logical.js

```diff
@@ -89,44 +94,54 @@ function insertLeaf(tree, leaf) {
   entry[leaf.name] = leaf;
 }
 
-function problem(root, issue) {
-  if (!root.problems) {
-    root.problems = [];
+function walkDeps(root, tree, from) {
+  if (!from) {
```

Comment:
We pass in the `from` since it *could* be different from the `tree.__from` property (when we're copying across a package's tree).